homeassistant/components/generic/config_flow.py

```diff
@@ -393,7 +393,8 @@ async def async_step_user_confirm_still(
                 title=self.title, data={}, options=self.user_input
             )
         register_preview(self.hass)
-        preview_url = f"/api/generic/preview_flow_image/{self.flow_id}?t={datetime.now().isoformat()}"
+        encoded_user_input = json.dumps(self.user_input, separators=(",", ":"))
+        preview_url = f"/api/generic/preview_flow_image/{self.flow_id}?d={encoded_user_input}&t={datetime.now().isoformat()}"
```

Comment:
Another option would be to keep the user_input internally (eg. `self.preview_cam`), and then use
```python
        flow = self.hass.config_entries.flow._progress.get(  # noqa: SLF001
            flow_id
        ) or self.hass.config_entries.options._progress.get(flow_id)  # noqa: SLF001
        if not flow:
            _LOGGER.warning("Unknown flow %s while getting image preview", flow_id)
            raise web.HTTPNotFound
        user_input = flow.preview_cam
```